stake-pool/js/test/instructions.test.ts

```diff
@@ -252,5 +261,60 @@ describe('StakePoolProgram', () => {
       expect(res.stakeReceiver).toEqual(undefined);
       expect(res.totalRentFreeBalances).toEqual(10000);
     });
+
+    it.only('withdraw to a stake account provided', async () => {
+      const stakeReceiver = new PublicKey(20);
+      connection.getAccountInfo = jest.fn(async (pubKey: PublicKey) => {
+        if (pubKey == stakePoolAddress) {
+          return stakePoolAccount;
+        }
+        if (pubKey.equals(CONSTANTS.poolTokenAccount)) {
+          return mockTokenAccount(LAMPORTS_PER_SOL * 2);
+        }
+        if (pubKey.equals(stakePoolMock.validatorList)) {
+          return mockValidatorList();
+        }
+        if (pubKey.equals(CONSTANTS.validatorStakeAccountAddress))
+          return mockValidatorsStakeAccount();
+        return null;
+      });
+      connection.getParsedAccountInfo = jest.fn(async (pubKey: PublicKey) => {
+        if (pubKey.equals(stakeReceiver)) {
+          return mockRpc(stakeAccountData);
+        }
+        return null;
+      });
+
+      const res = await withdrawStake(
+        connection,
+        stakePoolAddress,
+        tokenOwner,
+        1,
+        undefined,
+        undefined,
+        stakeReceiver,
+      );
+
+      expect((connection.getAccountInfo as jest.Mock).mock.calls.length).toBe(4);
+      expect((connection.getParsedAccountInfo as jest.Mock).mock.calls.length).toBe(1);
+      expect(res.instructions).toHaveLength(3);
+      expect(res.signers).toHaveLength(2);
+      expect(res.stakeReceiver).toEqual(stakeReceiver);
+      expect(res.totalRentFreeBalances).toEqual(10000);
+    });
+  });
+  describe('getStakeAccount', () => {
+    it.only('returns an uninitialized parsed stake account', async () => {
+      const stakeAccount = new PublicKey(20);
+      connection.getParsedAccountInfo = jest.fn(async (pubKey: PublicKey) => {
+        if (pubKey.equals(stakeAccount)) {
+          return mockRpc(uninitializedStakeAccount);
+        }
+        return null;
+      });
+      const parsedStakeAccount = await getStakeAccount(connection, stakeAccount);
+      expect((connection.getParsedAccountInfo as jest.Mock).mock.calls.length).toBe(1);
+      expect(parsedStakeAccount).toEqual(uninitializedStakeAccount.parsed);
+    });
```

Comment:
you were right so ```info``` needed to be optional to parse an ```Uninitialized``` stake account so added this test here don't know if ```instructions.test``` is the right place for this test to be, tell me if this needs to be placed somewhere else

Reply:
Seems fine to me!